File: src/NBdbm/Classes/Interfaces/iCOBR.vb

Namespace Interfaces.iCOBR

    Public Interface iCadastroAcionamentos
        Inherits System.IDisposable
        Sub Salvar(ByVal NoCommit As Boolean)
        Sub Salvar()
        Property Acionamento() As iCOBR.Primitivas.iAcionamentos
        ReadOnly Property ColecaoAcionamentos() As NBdbm.Fachadas.NbCollection
        ReadOnly Property ColecaoNovosAcionamentos() As NBdbm.Fachadas.NbCollection
        ReadOnly Property FichaDevedor() As NBdbm.Fachadas.plxCOBR.CadastroEntidade
    End Interface
    Public Interface iLacamentoBaixa
        Inherits System.IDisposable
        Sub Salvar(ByVal NoCommit As Boolean)
        Sub Salvar()
        Property Baixa() As iCOBR.Primitivas.iBaixas
    End Interface
    Namespace Primitivas
        Public Interface iTipoDivida
            Inherits Interfaces.Primitivas.iObjetoTabela
            Property Descricao() As String
        End Interface
        Public Interface iTipoAcionamento
            Inherits Interfaces.Primitivas.iObjetoTabela
            Property Descricao() As String
            Property DiasReacionamento() As Integer
            Property CredencialExigida() As Integer
        End Interface
        Public Interface iDivida
            Inherits Interfaces.Primitivas.iObjetoTabela
            Property idEntidade() As Integer
            Property idTipoDivida() As Integer
            Property Contrato() As String
            Property NumDoc() As Integer
            Property DataVencimento() As Date
            Property ValorNominal() As Double
            Property ValorNominalParcial() As Double
            Property DataBaixa() As Date
            Property BorderoBaixa() As Integer
            Property NumRecibo() As Integer
            Property idCobrador() As Integer
            Property PerCobrador() As Double
            Property idUsuarioBaixa() As Integer
            Property BaixaNoCliente() As Boolean
            Property Baixada() As Boolean
            Property BaixaParcial() As Boolean
            Property XmPathCliente() As String
        End Interface
        Public Interface iAcionamentos
            Inherits Interfaces.Primitivas.iObjetoTabela
            Property idEntidade() As Integer
            Property idUsuario() As Integer
            Property idTipoAcionamento() As Integer
            Property TextoRespeito() As String
            Property DataAcionamento() As System.DateTime
            Property DataPromessa() As Date
        End Interface
        Public Interface iTarifas
            Inherits Interfaces.Primitivas.iObjetoTabela
            Property idEntidade() As Integer
            Property Juros() As Double
            Property Multa() As Double
        End Interface
        Public Interface iBaixas
            Inherits Interfaces.Primitivas.iObjetoTabela
            Property idEntidade() As Integer
            Property idDivida() As Integer
            Property BaixadoCliente() As Boolean
            Property NumBordero() As Integer
            Property DataBaixa() As Date
            Property ValorNominal() As Double
            Property ValorBaixa() As Double
            Property ValorRecebido() As Double
        End Interface
    End Namespace
End Namespace

File: src/NBdbm/CS/Classes/Interfaces/iCOBR.cs
using System.Data;
using System.Collections;
using Microsoft.VisualBasic;
using System.Diagnostics;
using System;


namespace NBdbm.Interfaces.iCOBR
{


    public interface iCadastroAcionamentos : System.IDisposable
    {
        void Salvar(bool NoCommit);
        void Salvar();
        iCOBR.Primitivas.iAcionamentos Acionamento
        {
            get;
            set;
        }
        NBdbm.Fachadas.NbCollection ColecaoAcionamentos
        {
            get;
        }
        NBdbm.Fachadas.NbCollection ColecaoNovosAcionamentos
        {
            get;
        }
        NBdbm.Fachadas.plxCOBR.CadastroEntidade FichaDevedor
        {
            get;
        }
    }
    public interface iLacamentoBaixa : System.IDisposable
    {
        void Salvar(bool NoCommit);
        void Salvar();
        iCOBR.Primitivas.iBaixas Baixa
        {
            get;
            set;
        }
    }
    namespace Primitivas
    {
        public interface iTipoDivida : Interfaces.Primitivas.iObjetoTabela
        {
            string Descricao
            {
                get;
                set;
            }
        }
        public interface iTipoAcionamento : Interfaces.Primitivas.iObjetoTabela
        {
            string Descricao
            {
                get;
                set;
            }
            int DiasReacionamento
            {
                get;
                set;
            }
            int CredencialExigida
            {
                get;
                set;
            }
        }
        public interface iDivida : Interfaces.Primitivas.iObjetoTabela
        {
            int idEntidade
            {
                get;
                set;
            }
            int idTipoDivida
            {
                get;
                set;
            }
            string Contrato
            {
                get;
                set;
            }
            int NumDoc
            {
                get;
                set;
            }
            DateTime DataVencimento
            {
                get;
                set;
            }
            double ValorNominal
            {
                get;
                set;
            }
            double ValorNominalParcial
            {
                get;
                set;
            }
            DateTime DataBaixa
            {
                get;
                set;
            }
            int BorderoBaixa
            {
                get;
                set;
            }
            int NumRecibo
            {
                get;
                set;
            }
            int idCobrador
            {
                get;
                set;
            }
            double PerCobrador
            {
                get;
                set;
            }
            int idUsuarioBaixa
            {
                get;
                set;
            }
            bool BaixaNoCliente
            {
                get;
                set;
            }
            bool Baixada
            {
                get;
                set;
            }
            bool BaixaParcial
            {
                get;
                set;
            }
            string XmPathCliente
            {
                get;
                set;
            }
        }
        public interface iAcionamentos : Interfaces.Primitivas.iObjetoTabela
        {
            int idEntidade
            {
                get;
                set;
            }
            int idUsuario
            {
                get;
                set;
            }
            int idTipoAcionamento
            {
                get;
                set;
            }
            string TextoRespeito
            {
                get;
                set;
            }
            System.DateTime DataAcionamento
            {
                get;
                set;
            }
            DateTime DataPromessa
            {
                get;
                set;
            }
        }
        public interface iTarifas : Interfaces.Primitivas.iObjetoTabela
        {
            int idEntidade
            {
                get;
                set;
            }
            double Juros
            {
                get;
                set;
            }
            double Multa
            {
                get;
                set;
            }
        }
        public interface iBaixas : Interfaces.Primitivas.iObjetoTabela
        {
            int idEntidade
            {
                get;
                set;
            }
            int idDivida
            {
                get;
                set;
            }
            bool BaixadoCliente
            {
                get;
                set;
            }
            int NumBordero
            {
                get;
                set;
            }
            DateTime DataBaixa
            {
                get;
                set;
            }
            double ValorNominal
            {
                get;
                set;
            }
            double ValorBaixa
            {
                get;
                set;
            }
            double ValorRecebido
            {
                get;
                set;
            }
        }
    }
}
